### ISSUE TITLE
Attempted pointer arithmetic generates bad llvm IR

### ISSUE THREAD

**Issue by John-Titor** [bagel99/esl#6]
```
// pointer arithmetic generates bad llvm IR
proc test(arg: @[]_byte) : @[]_byte
{
	return arg + 2;
}
```
```
➜  issues git:(master) ✗ ../src/eslc2 -L9.0.0 -m x86-64 02_pointer_arithmetic.esl > out.ll
➜  issues git:(master) ✗ ../src/llvm -m x86-64 -I out.ll
/usr/local/opt/llvm/bin/opt: out.ll:9:11: error: invalid operand type for instruction
        %1 = add [0 x i8]* %0, inttoptr(i64 2 to [0 x i8]*)
                 ^
```

**Comment by bagel99** [comment]
I agree that's an ugly error message.
There is no explicit pointer arithmetic in ESL.  What you are doing is return a pointer to a sub-array:
proc test(arg: @[]_byte) : @[]_byte
{
	return arg[2:];
}

**Comment by John-Titor** [comment]
That's a much cleaner approach from the language perspective as well, thanks.

**Comment by bagel99** [comment]
Fix in commit 1a53f6c5568510630c47c4b1b93bee27a8d88679.
